### PR TITLE
Latest cuda model generator + example

### PR DESCRIPTION
**Pull request by DeLaVlag**
@maedoc @sdiazpier @liadomide This PR holds the latest cuda model generator and an example on how the produced model can be simulated on a GPU in which the parameters 'speed' and 'coupling' are explored doing a parameter sweep. Some restructuring is done mainly in the directory structure with the focus on integrating this in TVB root eventually. The dsl dir contains dsl_cuda and can eventually contain the dsl_regular model generator. In dsl_cuda an example is added in which a model is generated and run on a GPU based on the PyCuda library. The readme holds information on how to execute this example. The tvb/basic directory is added to overwrite global tvb configuration mainly for the logger such that it outputs its information on simulation.

### CODE REVIEW

**Comment by maedoc** [comment]
This is unreviewable with tvb.basic copy-pasted in here.  Can you remove that please?

File: dsl/dsl_cuda/tmpl8_CUDA.py [hunk 1]
```diff
@@ -126,7 +130,7 @@
     for (unsigned int t = i_step; t < (i_step + n_step); t++)
     {
     //***// This is the loop over nodes, which also should stay the same
-        for (unsigned int i_node = threadIdx.y; i_node < n_node; i_node+=blockDim.y)
```
Comment (maedoc) [inline dsl/dsl_cuda/tmpl8_CUDA.py:124]: did you notice any perf difference with unsigned vs signed?

Comment (DeLaVlag) [reply dsl/dsl_cuda/tmpl8_CUDA.py:124]: The reason for change is the index not the unsigned vs signed. Is there any indication that unsigned performs better (or should be used here)?

Comment (maedoc) [reply dsl/dsl_cuda/tmpl8_CUDA.py:124]: no sorry, I somehow misread the diff line

Comment (DeLaVlag) [reply dsl/dsl_cuda/tmpl8_CUDA.py:124]: np. this made me realize that I have to update the indexing as well. you can review but I have to push a commit still.

File: dsl/dsl_cuda/tmpl8_CUDA.py [hunk 2]
```diff
@@ -227,11 +244,11 @@
...
-            ##if (t == (i_step + n_step - 1)){
+            if (t == (i_step + n_step - 1)){
                 % for i, expo in enumerate(expolist):
                 tavg(i_node + ${i} * n_node) = ${expo};
                 % endfor /
-            ##}
+            }
 
             // sync across warps executing nodes for single sim, before going on to next time step
             __syncthreads();
```
Comment (maedoc) [inline dsl/dsl_cuda/tmpl8_CUDA.py:254]: I recall this being necessary for some optimisations where multiple threads were used inside a single simulation, but this is not the case here, so do we need `__syncthreads()`? Sorry if I'm not remembering something here.

**Pull request merged**